lib/src/runner/iojs_utils.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// io.js utils used by dartdevrun.
+library dev_compiler.src.runner.iojs_utils;
+
+import 'dart:io';
+import 'dart:async';
+
+import 'iojs_version.dart';
+
+/// io.js still has issues, like lexical this in arrow functions. 
+final _MIN_SUPPORTED_IOJS_VERSION =
+    new IoJsVersion.fromString("v3.0.0-next-nightly20150717cbec3ee19d");
+
+final _IOJS_BIN = "iojs";
+
+/// Violently exits if the io.js version in the path is not supported, or
+/// if there is no io.js in the path.
+Future checkIoJsVersion() async {
+  ProcessResult result = await Process.run(_IOJS_BIN, ["--version"]);
+  if (result.exitCode != 0) {
+    print("Failed to check the version of iojs (${result.stderr})");
+    exit(1);
+  }
+
+  IoJsVersion version = new IoJsVersion.fromString(result.stdout.trim());
+  if (version.compareTo(_MIN_SUPPORTED_IOJS_VERSION) < 0) {
+    print("The version of iojs in the path ($version) "
+        "is older than the required minimum $_MIN_SUPPORTED_IOJS_VERSION. "
+        "Please visit https://iojs.org/download/next-nightly/");
+    exit(1);
+  }
+  stderr.writeln("WARNING: io.js does not fully support some critical "
```

Comment:
feel free to restore the `=>` `this` workaround if you'd like:
https://github.com/dart-lang/dev_compiler/commit/19f6daf59b49e425961b01e34ebd50339f8f90a1#diff-3a4f8912d1b55456831b2e7b40ba8d68L847

should be pretty easy to pass an option down, so Printer only inserts the workaround if needed.
it was fixed a while ago in V8/Chrome, though, so maybe iojs will get it soon